sgkit/tests/test_grm.py

```diff
@@ -173,3 +174,745 @@ def test_genomic_relationship__raise_on_ancestral_frequency_missing():
+        [0.0, 0.0, 1.0],
+    ]
+    ds["pedigree_subset_inverse"] = ["genotypes_0", "genotypes_1"], [
+        [1.0, 0.0, 0.0],
+        [0.0, 1.0, 0.0],
+        [0.0, 0.0, 1.0],
+    ]
+    ds["genomic_sample_index"] = "genotypes", [1, 2]
+    with pytest.raises(
+        ValueError, match="The genomic matrix cannot be larger than the pedigree matrix"
+    ):
+        sg.hybrid_inverse_relationship(
+            ds,
+            pedigree_inverse_relationship="pedigree_inverse_relationship",
+            pedigree_subset_inverse_relationship="pedigree_subset_inverse",
+            genomic_inverse_relationship="genomic_inverse_relationship",
+            genomic_sample_index="genomic_sample_index",
+        )
+
+
+def test_hybrid_inverse_relationship__raise_on_subset_non_matching_dims():
+    ds = xr.Dataset()
+    ds["pedigree_inverse_relationship"] = ["samples_0", "samples_1"], [
+        [1.0, 0.0, 0.0],
+        [0.0, 1.0, 0.0],
+        [0.0, 0.0, 1.0],
+    ]
+    ds["genomic_inverse_relationship"] = ["genotypes_0", "genotypes_1"], [
+        [1.0, 0.0],
+        [0.0, 1.0],
+    ]
+    ds["pedigree_subset_inverse"] = ["others_0", "others_1"], [
+        [1.0, 0.0],
+        [0.0, 1.0],
+    ]
+    ds["genomic_sample_index"] = "genotypes", [1, 2]
+    message = "The dimensions of pedigree subset must match those of the genomic matrix"
+    with pytest.raises(ValueError, match=message):
+        sg.hybrid_inverse_relationship(
+            ds,
+            pedigree_inverse_relationship="pedigree_inverse_relationship",
+            pedigree_subset_inverse_relationship="pedigree_subset_inverse",
+            genomic_inverse_relationship="genomic_inverse_relationship",
+            genomic_sample_index="genomic_sample_index",
+        )
+
+
+def load_Legarra2009_example():
+    path = pathlib.Path(__file__).parent.absolute()
+    hrm = np.loadtxt(path / "test_grm/Legara2009_H_matrix.txt")
+    grm = np.loadtxt(path / "test_grm/Legara2009_G_matrix.txt")
+    ped = np.loadtxt(path / "test_grm/Legara2009_pedigree.txt").astype(int)
+    ds = xr.Dataset()
+    ds["hybrid_relationship"] = ["samples_0", "samples_1"], hrm
+    ds["genomic_relationship"] = ["genotypes_0", "genotypes_1"], grm
+    ds["genomic_inverse_relationship"] = [
+        "genotypes_0",
+        "genotypes_1",
+    ], np.linalg.inv(grm)
+    ds["genomic_sample_index"] = "genotypes", np.arange(8, 12)
+    ds["parent"] = ["samples", "parents"], ped
+    return ds
+
+
+@pytest.mark.parametrize("samples_chunks", [None, 10])
+@pytest.mark.parametrize("genotypes_chunks", [None, 2])
+@pytest.mark.parametrize("grm_position", ["default", "first", "last", "mixed"])
+def test_hybrid_relationship__Legarra(grm_position, genotypes_chunks, samples_chunks):
+    # simulate example from Legarra et al 2009
+    ds = load_Legarra2009_example()
+    ds = sg.pedigree_kinship(ds, return_relationship=True)
+    if grm_position == "first":
+        # sort so GRM samples are first
+        idx = [
```

Comment:
If you want the auto-formating to keep these on one line, remove the final comma.

Reply:
Thanks!